swagger2.0/oic.wk.p.swagger.json

```diff
@@ -1,222 +1,195 @@
-              "oic.if.rw",
-              "oic.if.r",
-              "oic.if.a",
-              "oic.if.s"
-            ],
-            "type": "string"
-          },
-          "minItems": 1,
-          "readOnly": true,
-          "type": "array"
-        }
-
-        }
-        ,"type" : "object"
-      ,"required": ["pi", "mnmn"]
-    }
-  }
-}
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Platform",
+    "version": "v1-20160622",
+    "license": {
+      "name": "OCF Data Model License",
+      "url": "https://github.com/openconnectivityfoundation/core/blob/e28a9e0a92e17042ba3e83661e4c0fbce8bdc4ba/LICENSE.md",
+      "x-copyright": "Copyright 2016-2019 Open Connectivity Foundation, Inc. All rights reserved."
+    },
+    "termsOfService": "https://openconnectivityfoundation.github.io/core/DISCLAIMER.md"
+  },
+  "schemes": ["http"],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "paths": {
+    "/oic/p" : {
+      "get": {
+        "description": "Known resource that is defines the platform on which an Server is hosted.\nAllows for platform specific information to be discovered.\nRetrieve the information about the Platform\n",
+        "parameters": [
+          {"$ref": "#/parameters/interface"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": {
+              "pi":   "54919CA5-4101-4AE4-595B-353C51AA983C",
+              "rt":   ["oic.wk.p"],
+              "mnmn": "Acme, Inc"
+            },
+            "schema": { "$ref": "#/definitions/Platform" }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "interface" : {
+      "in" : "query",
+      "name" : "if",
+      "type" : "string",
+      "enum" : ["oic.if.r", "oic.if.baseline"]
+    }
+  },
+  "definitions": {
+    "Platform" : {
+      "properties": {
+        "rt" : {
+          "description": "Resource Type of the Resource",
+          "items": {
+            "enum":   ["oic.wk.p"],
+            "type": "string"
+          },
+          "maxItems": 1,
+          "minItems": 1,
+          "uniqueItems": true,
+          "readOnly": true,
+          "type": "array"
+        },
+
+        "pi" : {
+          "allOf": [
+            {
+              "description": "Format pattern according to IETF RFC 4122.",
+              "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
+              "type": "string"
+            },
+            {
+              "description": "Platform Identifier",
+              "readOnly": true
+            }
+          ]
+        },
+
+        "mnfv" : {
+          "description": "Manufacturer's firmware version",
+          "maxLength": 64,
+          "readOnly": true,
+          "type": "string"
+        },
+
+        "vid" : {
+          "description": "Manufacturer's defined information for the platform. The content is freeform, with population rules up to the manufacturer",

```

Comment:
In the spec, the description uses the word "vendor" instead of "manufacturer".  Not a big deal, but it struck me as odd that "vid" stands for Vendor ID but the words Vendor and ID don't show up in the description here.